core/src/nodes/handled_node_tasks.rs

```diff
@@ -71,6 +72,7 @@ pub struct HandledNodesTasks<TInEvent, TOutEvent, THandler> {
 
 impl<TInEvent, TOutEvent, THandler> fmt::Debug for HandledNodesTasks<TInEvent, TOutEvent, THandler> {
     fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        f.debug_struct("HandledNodesTask").finish().expect("adding a string works");
```

Comment:
That's not how you do this. The two entries have to be merged into one.
However I don't like this change by itself. This struct is just a collection of tasks and thus can be printed the same way as a collection.

Reply:
> That's not how you do this. The two entries have to be merged into one.

Yeah, it's ugly for sure. Can you show me the better way? I didn't see an obvious way of doing it at the time.

I think it's odd that a struct doesn't include its own name in the debug impl. It was surprising to me when I saw it pop up in the logs and I had to spend time figuring out what it was I was looking for. Principle of least surprise etc.: I think the name should be there.

Reply:
> I didn't see an obvious way of doing it at the time.

Apologize, looks like there's no built-in way. I probably had memories of another library.

However the fact that there's no built-in way kind of pushes in favour of not doing that. `Vec`, `HashMap`, and all the collections in `std::collections` do the same as this code here, without having their name printed in front.

Reply:
This works and does the job:

```
        f.debug_struct("HandledNodesTasks")
            .field("tasks", &self.tasks.keys())
            .field("next_task_id", &self.next_task_id)
            .finish()
```

Yay or nay?